Assets/HoloToolkit/Utilities/Scripts/Solvers/Solver.cs

```diff
@@ -0,0 +1,409 @@
+        }
+
+        /// <summary>
+        ///   Typically when a solver becomes enabled, it should update its internal state to the system, in case it was disabled far away
+        /// </summary>
+        protected virtual void OnEnable()
+        {
+            //Ensure the Camera helper component exists
+            CameraMotionInfo cInfo = CameraCache.Main.gameObject.EnsureComponent<CameraMotionInfo>();
+
+            if (solverHandler != null)
+            {
+                SnapGoalTo(solverHandler.GoalPosition, solverHandler.GoalRotation);
+            }
+
+            lifetime = 0;
+        }
+
+        protected virtual void Start()
+        {
+            //TransformTarget overrides ObjectToReferenceEnum
+            if (!solverHandler.TransformTarget)
+            {
+                StartCoroutine(CoStart());
+            }
+        }
+
+        protected IEnumerator CoStart()
+        {
+                switch (solverHandler.TrackedObjectToReference)
+                {
+                    case SolverHandler.TrackedObjectToReferenceEnum.Head:
+                        while (CameraCache.Main == null && CameraCache.Main.transform == null)
+                        {
+                            yield return null;
+                        }
+                        //Base transform target to camera transform
+                        solverHandler.TransformTarget = CameraCache.Main.transform;
+                        break;
+
+                    case SolverHandler.TrackedObjectToReferenceEnum.MotionControllerLeft:
+                        if(GetComponent<ControllerFinder>() == null)
+                        {
+                            Debug.LogError("There is no component SolverControllerFinder on " + this.gameObject.name);
+                            break;
+                        }
+                        GetComponent<ControllerFinder>().Handedness = InteractionSourceHandedness.Left;
+                        GetComponent<ControllerFinder>().OnEnable();
+                        while (GetComponent<ControllerFinder>().ElementTransform == null)
+                        {
+                            yield return null;
+                        }
+                        //Base transform target to Motion controller transform
+                        solverHandler.TransformTarget = this.GetComponent<ControllerFinder>().ElementTransform;
+                        break;
+
+                    case SolverHandler.TrackedObjectToReferenceEnum.MotionControllerRight:
+                        if (GetComponent<ControllerFinder>() == null)
+                        {
+                            Debug.LogError("There is no component SolverControllerFinder on " + this.gameObject.name);
+                            break;
+                        }
+                        GetComponent<ControllerFinder>().Handedness = InteractionSourceHandedness.Right;
+                        GetComponent<ControllerFinder>().OnEnable();
```

Comment:
Does this call work anymore with the change to `protected`? Though really, I feel like this pattern should be reworked. Calling a script's `OnEnable` doesn't seem like the best way to do things. Can `SolverHandler` be a `ControllerFinder`, that way we don't have to keep track of handedness in two places (`SolverHandler.TrackedObjectToReferenceEnum.MotionControllerRight` and `ControllerFinder.Handedness`)? Or perhaps `ControllerFinder` can have some events this script can register for, like `AttachToController` has.